### PR TITLE
Fix form validation for password

### LINKED ISSUE #176: Need signup error reporting
Signup page needs to report errors when signup call fails because of bad password, duplicate email/username, etc.

Found during QA in #11

### PR DESCRIPTION
**Pull request by alicoding**
Closes #176

### CODE REVIEW

**Comment by alicoding** [comment]
@secretrobotron r?

File: templates/lib/api.jsx [hunk 1]
```diff
@@ -48,5 +48,53 @@ module.exports = {
...
+    if(!caseValid) {
+      WebmakerActions.displayError({'field': 'password', 'message': 'Password must contain at least one uppercase and lowercase letter.'});
+    }
+    if(!digitValid) {
+      WebmakerActions.displayError({'field': 'password', 'message': 'Password must contain at least one number.'});
+    }
+    if(username) {
+      var containUserValid = !password.match(username);
+      if(!containUserValid) {
+        WebmakerActions.displayError({'field': 'password', 'message': 'Password cannot contain your username'});
```
Comment (secretrobotron) [inline templates/lib/api.jsx]: Missing a period

**Comment by secretrobotron** [comment]
Ran into a problem. Try this:
1. Go to /signup.
2. Enter a bad password.
3. Press tab to get it to show you the error.
4. Go back to the password textbox and type in a valid password.
5. Press tab again. Error should disappear, but it stays.

File: templates/pages/reset-password.jsx [hunk 2]
```diff
@@ -31,6 +31,7 @@ var ResetPassword = React.createClass({
     linkQuery.state = this.state.queryObj.state;
     linkQuery.scopes = this.state.queryObj.scopes;
     linkQuery.response_type = this.state.queryObj.response_type;
+    console.log(this.state.queryObj.uid)
```
Comment (cadecairos) [inline templates/pages/reset-password.jsx]: EXTERMINATE

**Pull request merged**